nrf_provisioning integration into nrf_cloud_rest_device_message sample (#12567)
nrfconnect/sdk-nrf: samples/cellular/nrf_cloud_rest_device_message/README.rst

The sample sends an alert when the device first comes online.
It also sends a log message indicating the sample version, as well as when the button is pressed.

> The sample can also be configured to use the `nRF Cloud Provisioning Service`_ with the :ref:`lib_nrf_provisioning` library.

Reviewer: ```suggestion
You can also configure the sample to use the `nRF Cloud Provisioning Service`_ with the :ref:`lib_nrf_provisioning` library.
```

User interface
**************

> Once the device is provisioned and connected, each press of button 1 generates a device-to-cloud button press `Device Message <nRF Cloud Device Messages_>`_ over REST.

Reviewer: ```suggestion
Once the device is provisioned and connected, each press of **Button 1** generates a device-to-cloud button press `Device Message <nRF Cloud Device Messages_>`_ over REST.
```

Set the :kconfig:option:`CONFIG_NRF_CLOUD_LOG_OUTPUT_LEVEL` option to the log level of messages to send to nRF Cloud, such as ``4`` for debug log messages.
Set the :kconfig:option:`CONFIG_NRF_CLOUD_REST_DEVICE_MESSAGE_SAMPLE_LOG_LEVEL_DBG` option so that log messages are generated on each button press.

To enable remote provisioning with the `nRF Cloud Provisioning Service`_ add the following parameter to your build command:

``-DOVERLAY_CONFIG=overlay-nrf_provisioning.conf``

This overlay enables the :ref:`lib_nrf_provisioning` library and its provisioning shell.
The sample will periodically check for provisioning commands.

> Press button 2 to manually initiate a check for provisioning commands.

Reviewer: ```suggestion
Press **Button 2** to manually initiate a check for provisioning commands.
```
